### PR TITLE
WOR-888 protected data policy

### PR DESCRIPTION
**Pull request by dvoet**
added a generic `PolicyLabel` class, used if for `protected-data`, reorganized constants a little bit and added some tests

### CODE REVIEW

File: service/src/main/java/bio/terra/policy/service/policy/PolicyLabel.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  /**
+   * Return either dependent or source, whichever is non-null, or null if both are null.
+   *
+   * @param dependent policy input
+   * @param source policy input
+   * @return policy input or null, if there is a conflict
+   */
+  @Override
+  public PolicyInput combine(PolicyInput dependent, PolicyInput source) {
+    return dependent == null ? source : dependent;
```
Comment (mikenev) [inline service/src/main/java/bio/terra/policy/service/policy/PolicyLabel.java:28]: Would this work to combine 2 different PolicyLabels? We're adding protected-data right now, if there's another type then we should be able to combine them. Maybe this is fine for now and we can cross that bridge when we add another label type.

Comment (ddietterich) [reply service/src/main/java/bio/terra/policy/service/policy/PolicyLabel.java:28]: It would be a good defense to verify that the policy name matches.

**Review by ddietterich — APPROVED** [review]
Suggest a fix to the `combine` method. Otherwise, LGTM

**Comment by dvoet** [comment]
@mikenev @ddietterich I went a step further that the PR feedback and made sure _all_ policy classes operate only on the policies they are meant to. I did this within `PolicyBase` turning it into an abstract class.

File: service/src/test/java/bio/terra/policy/service/policy/PolicyGroupConstraintTest.java [hunk 2]
```diff
@@ -77,11 +76,10 @@ void groupConstraintTest_combineEmptyDestination() throws Exception {
   void groupConstraintTest_combineEmptySourceAndDestination() throws Exception {
     var groupConstraint = new PolicyGroupConstraint();
 
-    // Only adding region constraints.
     var dependentPolicy =
-        new PolicyInput(TERRA_NAMESPACE, REGION_CONSTRAINT, buildMultimap(REGION_KEY, GROUP_NAME));
+        new PolicyInput(TERRA_NAMESPACE, GROUP_CONSTRAINT, buildMultimap(REGION_KEY));
     var sourcePolicy =
-        new PolicyInput(TERRA_NAMESPACE, REGION_CONSTRAINT, buildMultimap(REGION_KEY, GROUP_NAME));
+        new PolicyInput(TERRA_NAMESPACE, GROUP_CONSTRAINT, buildMultimap(REGION_KEY));
```
Comment (dvoet) [inline service/src/test/java/bio/terra/policy/service/policy/PolicyGroupConstraintTest.java:82]: this was actually an invalid test as written, using a group constraint policy to combine region constraint policies

**Review by ddietterich — APPROVED** [review]
Cool! Thanks, Doug

**Pull request merged**